diff --git a/node_modules/@angular/router/router_module.d-Bx9ArA6K.d.ts b/node_modules/@angular/router/router_module.d-Bx9ArA6K.d.ts
index 0f635d1..38bc37c 100755
--- a/node_modules/@angular/router/router_module.d-Bx9ArA6K.d.ts
+++ b/node_modules/@angular/router/router_module.d-Bx9ArA6K.d.ts
@@ -1764,7 +1764,7 @@ declare class ActivatedRoute {
     /** The component of the route, a constant. */
     component: Type<any> | null;
     /** The current snapshot of this route */
-    snapshot: ActivatedRouteSnapshot;
+    snapshot?: ActivatedRouteSnapshot;
     /** An Observable of the resolved route title */
     readonly title: Observable<string | undefined>;
     /** An observable of the URL segments matched by this route. */
