diff --git a/node_modules/@angular/router/types/_router_module-chunk.d.ts b/node_modules/@angular/router/types/_router_module-chunk.d.ts
index 7b718da..520b381 100755
--- a/node_modules/@angular/router/types/_router_module-chunk.d.ts
+++ b/node_modules/@angular/router/types/_router_module-chunk.d.ts
@@ -495,7 +495,7 @@ declare class RouteConfigLoadEnd {
  */
 declare class ChildActivationStart {
     /** @docsNotRequired */
-    snapshot: ActivatedRouteSnapshot;
+    snapshot?: ActivatedRouteSnapshot;
     readonly type = EventType.ChildActivationStart;
     constructor(
     /** @docsNotRequired */
